move-to-sdk/app/io/sphere/sdk/play/metrics/MetricAction.java

```diff
@@ -44,6 +45,7 @@ private void logRequestDataOnComplete(final Http.Context ctx, final List<ReportR
     }
 
     private Integer calculateTotalSize(final List<ReportRawData> rawDatas) {
-        return rawDatas.stream().map(elem -> elem.getHttpResponse().getResponseBody().map(b -> b.length).orElse(0)).reduce(0, (a, b) -> a + b);
+        return rawDatas.stream().map(elem -> Optional.ofNullable(elem.getHttpResponse().getResponseBody())
+                .map(b -> b.length).orElse(0)).reduce(0, (a, b) -> a + b);
```

Comment:
I would use https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#mapToInt-java.util.function.ToIntFunction-

and then just call 

https://docs.oracle.com/javase/8/docs/api/java/util/stream/IntStream.html#sum--